torch/amp/grad_scaler.py

```diff
@@ -334,6 +343,9 @@ def unscale_(self, optimizer: torch.optim.Optimizer) -> None:
         assert self._scale is not None
         inv_scale = self._scale.double().reciprocal().float()
         found_inf = torch.full((), 0.0, dtype=torch.float32, device=self._scale.device)
+        if optimizer.param_groups[0]["params"] is not None:
+            for grad_tensor in optimizer.param_groups[0]["params"]:
+                found_inf = grad_tensor.new_full((), 0.0)
```

Comment:
If I understand correctly, this could should run only once, but now runs in a forloop. Should this instead be run on just the first param in the optimizer?

I'm also guessing the goal here is to create a Tensor subclass. It would be good to include a comment with that in mind, otherwise it is weird for the grad_scaler to randomly access a parameter in the optimizer.

Reply:
Running once is good, we can follow the first tensor type in grads.
Comments will be added in the next version.